File: VB/WinSolution.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports System

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _createdOn As DateTime
		<Custom("DisplayFormat", "{0: ddd, dd MMMM yyyy hh:mm:ss tt}"), Custom("EditMask", "ddd, dd MMMM yyyy hh:mm:ss tt"), Custom("PropertyEditorType", "WinSolution.Module.Win.MyDatePropertyEditor")> _
		Public Property CreatedOn() As DateTime
			Get
				Return _createdOn
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("CreatedOn", _createdOn, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/E1497/WinSolution.Module/DomainObject1.cs
using DevExpress.ExpressApp.Model;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using System;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private DateTime _createdOn;
        [ModelDefault("DisplayFormat", "{0: ddd, dd MMMM yyyy hh:mm:ss tt}")]
        [ModelDefault("EditMask", "ddd, dd MMMM yyyy hh:mm:ss tt")]
        [ModelDefault("PropertyEditorType", "WinSolution.Module.Win.MyDatePropertyEditor")]
        public DateTime CreatedOn {
            get { return _createdOn; }
            set { SetPropertyValue("CreatedOn", ref _createdOn, value); }
        }
    }
}
